integration_test.go

```diff
@@ -368,31 +372,273 @@ func (s *simpleService) Baz(c context.Context, in *simple.Foo) (*simple.Foo, err
 	return nil, fmt.Errorf("negative input")
 }
 
-func (s *simpleService) BidiStream(stream simple.Bar_BidiStreamServer) error {
-	return nil
+func (s *simpleService) ClientStream(stream simple.Bar_ClientStreamServer) error {
+	if s.exitOnStart {
+		return nil
+	}
+	if s.errOnStart != nil {
+		return s.errOnStart
+	}
+	counter := 0
+	for {
+		_, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		counter++
+	}
+	return stream.SendAndClose(&simple.Foo{Test: int32(counter)})
 }
 
-func (s *simpleService) ClientStream(stream simple.Bar_ClientStreamServer) error {
+func (s *simpleService) ServerStream(req *simple.Foo, stream simple.Bar_ServerStreamServer) error {
+	if s.errOnStart != nil {
+		return s.errOnStart
+	}
+	for i := 0; i < int(req.Test); i++ {
+		if err := stream.Send(&simple.Foo{Test: int32(i + 1)}); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
-func (s *simpleService) ServerStream(req *simple.Foo, stream simple.Bar_ServerStreamServer) error {
+func (s *simpleService) BidiStream(stream simple.Bar_BidiStreamServer) error {
+	if s.exitOnStart {
+		return nil
+	}
+	if s.errOnStart != nil {
+		return s.errOnStart
+	}
+	for {
+		msg, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		msg.Test += 100
+		if err = stream.Send(msg); err != nil {
+			return err
+		}
+	}
```

Comment:
What do you think of simplifying all of this handler by having in each request message a field `expectedReturnValue`? The big benefits of having this is that for maintaining the test we will not need to understand the logic of the handler, but everything will be driven by the test itself.

This way of doing could also apply for the field `errOnStart` and `exitOnStart`.

Thoughts?

Reply:
+1

Added the simple expected input and output in this test